code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -1025,6 +1055,21 @@ pub mod pallet {
 					// position, that way any rounding is accounted for.
 					let new_stake = left_from_one_ratio.mul_ceil(existing_position.stake);
 					let new_share = left_from_one_ratio.mul_ceil(existing_position.share);
+
+					let rewards_pool = RewardPools::<T>::get(existing_position.reward_pool_id)
```

Comment:
More robust solution with less code in transactional extrinsic would be fail on save to storage. So any attempt to save stake with low amount could be done via one API call, regardless if that was split, stake, unstake. Error will be StakeBalanceTooLow or something. `StakedAmountTooLowAfterSplit` encodes 2 things, `she called split and get to low balance`. But she already know what tx she called. Error is not common path and increased weight in case of error is fine because it will not be larger then when success happens.

So I want to tell is that code is more complicated then it can be.

Reply:
I think this current implementation is fine for now, we can look into possible alternatives in the future